src/System.Text.Primitives/System/Text/TextEncoding.cs

```diff
@@ -2,13 +2,293 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.ComponentModel;
+using System.Text.Utf8;
 
 namespace System.Text {
 
-    public enum TextEncoding : byte
+    public abstract class TextEncoder
     {
```

Comment:
Yeah, I will rename in a subsequent PR. I diod not want to rename it in this one as there were so many changes to the file that git would most probably not detect it as a rename but rather delete + new file.